zerver/tests/test_digest.py

```diff
@@ -34,25 +34,16 @@ def test_multiple_stream_senders(self,
         cutoff = time.mktime(one_sec_ago.timetuple())
 
         senders = ['hamlet', 'cordelia',  'iago', 'prospero', 'ZOE']
-        for sender_name in senders:
-            email = self.example_email(sender_name)
-            self.login(email)
-            content = 'some content for ' + email
-            payload = dict(
-                type='stream',
-                client=client,
-                to='Verona',
-                topic='lunch',
-                content=content,
-            )
-            result = self.client_post("/json/messages", payload)
-            self.assert_json_success(result)
+        self.simulate_stream_conversation('Verona', senders)
 
         flush_per_request_caches()
         with queries_captured() as queries:
             handle_digest_email(othello.id, cutoff)
 
-        self.assertTrue(21 <= len(queries) <= 22)
+        # One additional query is run to the content_type for the
+        # UserProfile model, when this test is run in isolation, as opposed to
+        # the whole class or test-suite.
+        self.assertTrue(6 <= len(queries) <= 7)
```

Comment:
We can probably make this a fixed number by just doing whatever triggers that query the ling before the `with` block.

Reply:
Agree! :+1: